### PR TITLE
Laushinka/server logs incoming 9556

### LINKED ISSUE #9556: Server logs incoming requests and their status codes
Log incoming requests.

### PR DESCRIPTION
**Pull request by laushinka**
## Description
Implements a middleware to log incoming requests.
This is an initial PR to prep for a follow-up PR that implements an external middleware library that has what we need out of the box.

## Related Issue(s)
Fixes #9556 

## How to test
Under `/component/public-api-server` run `go test ./...`

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@laushinka: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/public-api-server/middleware/logging_test.go
Comment (laushinka) [inline components/public-api-server/middleware/logging_test.go:26]: @easyCZ I'm having trouble injecting the logger here. It's expecting type `*logrus.Entry` but I couldn't figure out how to pass that in 🙏🏽

**Comment by laushinka** [comment]
@easyCZ Ready for review!

File: components/public-api-server/middleware/logging.go
Comment (easyCZ) [inline components/public-api-server/middleware/logging.go:25]:
```suggestion
			}).Infof("Handled HTTP request %s %s", method, uri)
```

By also adding the method and URI into the log, we can more easily visually grep and we still have it in the structured fields for deeper analysis. 
It also changes from `succesful` because that's somewhat ambigous. The fact we handled it doesn't mean we returned a 200 (or something that can be consider a success) so that's the reasoning for the text change.

**Review by easyCZ — APPROVED** [review]
LGTM, modulo the log message change. Adding hold to leave it up to you if you change it in this PR or another.
/hold

File: components/public-api-server/middleware/logging.go
Comment (laushinka) [reply components/public-api-server/middleware/logging.go:25]: This makes sense, thanks!

**Pull request merged**
